tests-legacy/Unit/ContextMocker.php

```diff
@@ -111,10 +112,9 @@ public function mockContext()
         Module::setContextInstanceForTesting($context);
         $context->shop = new Shop((int) Configuration::get('PS_SHOP_DEFAULT'));
         Shop::setContext(Shop::CONTEXT_SHOP, (int) Context::getContext()->shop->id);
-        $context->customer = Phake::mock('Customer');
-        Phake::when($context->customer)->getGroups()->thenReturn(array());
-        $context->cookie   = Phake::mock('Cookie');
-        $context->country  = Phake::mock('Country');
+        $context->customer = new Customer();
```

Comment:
Waaaah 😅  this class is complex.

So before your PR we had a mix of mocks, real ObjectModels, custom mock/stub ... such a crazy mix !

Since this is inside `tests-legacy` folder, I think your idea is good: better to follow a single way of setup a legacy environment and use real ObjectModels.

I am OK for either full real objects or full mocks, but mix might create a complex situation hard to maintain IMO.

I approve the PR 👍 

FYI
- phake usage was introduced in https://github.com/PrestaShop/PrestaShop/pull/8364/
- @tomlev if you happen to be around, do you remember this PR and do you have some warnings/highlights about why it would be a bad idea to stop using phake here?